backend/app/rest/api/rest_private.go

```diff
@@ -314,6 +317,55 @@ func (s *private) sendEmailConfirmationCtrl(w http.ResponseWriter, r *http.Reque
 	render.JSON(w, r, R.JSON{"user": user, "address": address})
 }
 
+// sendTelegramConfirmationCtrl gets address and siteID from query, makes confirmation token and sends it to user.
+// GET /telegram/subscribe?site=siteID&address=@someone
+//nolint:dupl // too hard to deduplicate that logic, as then it's tricky to use SendErrorJSON
+func (s *private) sendTelegramConfirmationCtrl(w http.ResponseWriter, r *http.Request) {
+	user := rest.MustGetUserInfo(r)
+	address := r.URL.Query().Get("address")
+	siteID := r.URL.Query().Get("site")
+	if address == "" {
+		rest.SendErrorJSON(w, r, http.StatusBadRequest,
+			errors.New("missing parameter"), "address parameter is required", rest.ErrInternal)
+		return
+	}
+	existingAddress, err := s.dataService.GetUserTelegram(siteID, user.ID)
```

Comment:
btw, if i understant "tricky to use SendErrorJSON" part this is about incorrect frame report? this can be solved with an extra param "skipFrames" of some sort. Not saying we have to refactor this, just a note

Reply:
Nope, just we'll need to have a nested function which also could potentially send an answer to the user (SendErrorJSON) so it would need request and response passed to it, and some boolean flag to check if we can carry on with our logic or that function took care about it for us and we need to just return immediately. It will hurt readability more than code duplication in my opinion.